sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -968,39 +964,37 @@ def do_stored_procedure_check(self):
         custom_tags = self.instance.get("tags", [])
 
         if (guardSql and self.proc_check_guard(guardSql)) or not guardSql:
-            self.connection.open_db_connections(self.connection.DEFAULT_DB_KEY)
-            cursor = self.connection.get_cursor(self.connection.DEFAULT_DB_KEY)
-
-            try:
-                self.log.debug("Calling Stored Procedure : %s", proc)
-                if self.connection.connector == 'adodbapi':
-                    cursor.callproc(proc)
-                else:
-                    # pyodbc does not support callproc; use execute instead.
-                    # Reference: https://github.com/mkleehammer/pyodbc/wiki/Calling-Stored-Procedures
-                    call_proc = '{{CALL {}}}'.format(proc)
-                    cursor.execute(call_proc)
-
-                rows = cursor.fetchall()
-                self.log.debug("Row count (%s) : %s", proc, cursor.rowcount)
-
-                for row in rows:
-                    tags = [] if row.tags is None or row.tags == '' else row.tags.split(',')
-                    tags.extend(custom_tags)
+            with self.connection.open_managed_default_connection():
+                with self.connection.get_managed_cursor() as cursor:
+                    try:
+                        self.log.debug("Calling Stored Procedure : %s", proc)
+                        if self.connection.connector == 'adodbapi':
+                            cursor.callproc(proc)
+                        else:
+                            # pyodbc does not support callproc; use execute instead.
+                            # Reference: https://github.com/mkleehammer/pyodbc/wiki/Calling-Stored-Procedures
+                            call_proc = '{{CALL {}}}'.format(proc)
+                            cursor.execute(call_proc)
```

Comment:
Is this change required for this PR? if it's a cleanup, let's open another PR to minimize the changes